int32  index
int32  matrix
string offset
---
int32  res